lib/kitchen/driver/ec2.rb

```diff
@@ -194,84 +247,176 @@ def default_aws_session_token
+          :sleep => config[:retryable_sleep],
+          :on => TimeoutError
+        ) do |retries, exception|
+          c = retries*config[:retryable_sleep]
+          t = config[:retryable_tries]*config[:retryable_sleep]
+          info "Waited #{c}/#{t} for spot request <#{spot_request_id}> to become fulfilled."
+          server = ec2.instances.filter('spot-instance-request-id', spot_request_id).to_a[0]
+          raise TimeoutError if server.nil?
+        end
+        info("Instance <#{server.id}> requested.")
+        tag_server(server)
+      end
+
+      def tag_server(server)
+        # tag assignation on the instance.
+        config[:tags].each do |k, v|
+          server.tag(k, :value => v)
+        end
+        server
+      end
+
+      def ec2_instance_data
+        i = {
+          :availability_zone            => config[:availability_zone],
+          :instance_type                => config[:instance_type],
+          :ebs_optimized                => config[:ebs_optimized],
+          :image_id                     => config[:image_id],
+          :key_name                     => config[:aws_ssh_key_id],
+          :subnet                       => config[:subnet_id],
+          :iam_instance_profile         => config[:iam_profile_name],
+          :associate_public_ip_address  => config[:associate_public_ip]
+        }
+        i[:block_device_mappings] = block_device_mappings unless block_device_mappings.empty?
+        i[:security_group_ids] = config[:security_group_ids] if config[:security_group_ids]
+        i[:user_data] = prepared_user_data if prepared_user_data
+        i
       end
 
       def debug_server_config
-        debug("ec2:region '#{config[:region]}'")
-        debug("ec2:availability_zone '#{config[:availability_zone]}'")
-        debug("ec2:flavor_id '#{config[:flavor_id]}'")
-        debug("ec2:ebs_optimized '#{config[:ebs_optimized]}'")
-        debug("ec2:image_id '#{config[:image_id]}'")
-        debug("ec2:private_ip_address '#{config[:private_ip_address]}'")
-        debug("ec2:security_group_ids '#{config[:security_group_ids]}'")
-        debug("ec2:tags '#{config[:tags]}'")
-        debug("ec2:key_name '#{config[:aws_ssh_key_id]}'")
-        debug("ec2:subnet_id '#{config[:subnet_id]}'")
-        debug("ec2:iam_profile_name '#{config[:iam_profile_name]}'")
-        debug("ec2:associate_public_ip '#{config[:associate_public_ip]}'")
-        debug("ec2:user_data '#{config[:user_data]}'")
-        debug("ec2:ssh_timeout '#{config[:ssh_timeout]}'")
-        debug("ec2:ssh_retries '#{config[:ssh_retries]}'")
-        debug("ec2:spot_price '#{config[:price]}'")
+        debug('EC2 Server Configuration')
+        names = [
+          :region, :availability_zone, :instance_type, :ebs_optimized, :image_id,
+          :private_ip_address, :security_group_ids, :tags, :aws_ssh_key_id, :subnet_id,
+          :iam_profile_name, :associate_public_ip, :user_data, :price
+        ]
```

Comment:
I agree - kitchen diagnose is a much better pattern, but as you have said, lets take it in steps.